### ISSUE TITLE
Error: response not received when handling concurrent telnet connections to multiple OLTs

### ISSUE THREAD

**Issue by rickyraz** [mkozjak/node-telnet-client#245]
## Description
When running telnet-client with concurrent connections to multiple OLTs in a scheduled manner, we're encountering frequent `Error: response not received` errors. This occurs in a production environment where we're managing connections to 120 OLTs, executing in batches of 5 concurrent connections every 25 minutes.

## Environment 
- Node.js version: node:22-alpine
- node-telnet-client version: 2.2.1
- Operating System: Ubuntu 22.04.5 LTS

## Current Behavior
The error occurs randomly during execution:
```
Error: response not received
    at Timeout._onTimeout (/app/node_modules/telnet-client/src/index.ts:401:26)
    at listOnTimeout (node:internal/timers:594:17)
    at process.processTimers (node:internal/timers:529:7)
```

Key characteristics of the issue:
- Occurs intermittently
- Happens more frequently when managing multiple concurrent connections
- Error appears to be timing-related
- Affects production workload managing 120 OLTs

## Expected Behavior
- Successful telnet connections to all OLTs
- Proper handling of timeouts or connection issues
- Stable operation across multiple scheduling cycles

## Additional Context
- The error seems to occur more frequently after several cycles of the scheduler
- We're using a concurrent batch size of 5 to avoid overwhelming the network
- Each cycle runs every 25 minutes
- The issue appears to be related to the timeout handling in the telnet-client library

Would you be able to provide guidance on:
1. Is this a known issue?
2. Are there recommended timeout or configuration settings for high-concurrency scenarios?
3. Is there a recommended approach for handling multiple telnet connections in a scheduled manner?

**Comment by mkozjak** [comment]
Hello,

`The issue appears to be related to the timeout handling in the telnet-client library`
Have you tried tcpdumping the whole traffic seeing if the telnet server responds back with data? If the response is empty, the client throws the message you sent.

**Comment by mkozjak** [comment]
Not really, because we have to be 100% sure the issue is not on your equipment before trying to debug the library.

**Comment by rickyraz** [comment]
Thank you for your advice, I will come back if I believe it's because of the device
